code_generator/GdkCodeGenerator/Templates/UnityTypeContent.tt

```diff
@@ -11,6 +11,21 @@ public struct <#= typeDetails.CapitalisedName #>
     public <#= fieldDetails.Type #> <#= fieldDetails.PascalCaseName #>;
 <# } #>
 
+
+<#
+if (fieldDetailsList.Count > 0)
+{ 
+    var constructorArgs = GetConstructorArgs();
+#>
+    public <#= typeDetails.CapitalisedName #>(<#= constructorArgs #>)
+    {
+<# foreach (var fieldDetails in fieldDetailsList) { #>
+        <#= fieldDetails.PascalCaseName #> = <#= fieldDetails.CamelCaseName #>;
+<# } #>
+    }
+<# } #>
+
```

Comment:
do we want to move the empty lines inside the conditional to avoid generating new lines everywhere?

Reply:
Probably - will review this